tests/knowledge_base/test_actions.py

```diff
@@ -152,3 +193,20 @@ async def test_action_run(data_file, slots, expected_slots):
             actual_msg = dispatcher.messages[0]["text"]
 
             assert actual_msg == expected_msg
+
+    # Check that temporary slot setting by action is correct.
+    if not any(slots):
+        object_types = knowledge_base.get_object_types()
+        object_type = match_extracted_entities_to_object_type(tracker, object_types)
+        set_object_type_slot_event = [SlotSet(SLOT_OBJECT_TYPE, object_type)]
+        tracker.add_slots(set_object_type_slot_event)
+        if slots.get(SLOT_MENTION) is None:
+            expected_temp_object_type_value = "restaurant"
+            actual_temp_object_type_value = tracker.slots[SLOT_OBJECT_TYPE]
+
+            assert actual_temp_object_type_value == expected_temp_object_type_value
+        else:
+            expected_temp_object_type_value = None
+            actual_temp_object_type_value = tracker.slots[SLOT_OBJECT_TYPE]
+
+            assert actual_temp_object_type_value == expected_temp_object_type_value
```

Comment:
The test shouldn't repeat the functionality that is executed by `action.run()`, but actually check if the instance of Tracker used by `action.run()` has the `SlotSet(SLOT_OBJECT_TYPE, object_type)` event in its `events` attribute before `SLOT_OBJECT_TYPE` gets reset to None. Is that possible?

Reply:
Thanks for clarifying that Anca Lita. In that case, we have used `tracker.events` to assert whether `SlotSet(SLOT_OBJECT_TYPE, object_type)` event is passed as expected during `action.run()`.